test/standby_test.go

```diff
@@ -0,0 +1,307 @@
+			if err != nil {
+				log.Println("Error while getting StatefulSet resource : ", err)
+				continue
+			}
+			if statefulSet.Status.AvailableReplicas == 1 {
+				return
+			}
+		case <-time.After(resourceConfigs.TestTimeout):
+			log.Println("Timeout while waiting for StatefulSet to be ready")
+			return
+		}
+	}
+}
+func (r *StandByTest) givenBackupPvcIsCreated() {
+	r.resourceCreator.CreateBackUpPvc()
+}
+
+func (r *StandByTest) givenExistingKubegresSpecIsSetTo(specNbreReplicas int32) {
+	var err error
+	r.kubegresResource, err = r.resourceRetriever.GetKubegres()
+
+	if err != nil {
+		log.Println("Error while getting Kubegres resource : ", err)
+		Expect(err).Should(Succeed())
+		return
+	}
+
+	r.kubegresResource.Spec.Replicas = &specNbreReplicas
+}
+
+func (r *StandByTest) givenKubegresSpecIsSetToBackup(backupSchedule, backupPvcName, backupVolumeMount string, specNbreReplicas int32) {
+	if backupSchedule != "" {
+		r.kubegresResource.Spec.Backup.Schedule = backupSchedule
+		r.kubegresResource.Spec.Backup.PvcName = backupPvcName
+		r.kubegresResource.Spec.Backup.VolumeMount = backupVolumeMount
+	}
+}
+
+func (r *StandByTest) whenKubegresIsCreated() {
+	r.resourceCreator.CreateKubegres(r.kubegresResource)
+}
+
+func (r *StandByTest) whenKubegresIsUpdated() {
+	r.resourceCreator.UpdateResource(r.kubegresResource, "Kubegres")
+}
+func (r *StandByTest) thenErrorEventShouldBeLogged() {
+	expectedErrorEvent := util.EventRecord{
+		Eventtype: v12.EventTypeWarning,
+		Reason:    "SpecCheckErr",
+		Message:   "In the Resources Spec the value of 'spec.standby.primaryEndpoint' is undefined. Please set a value otherwise this operator cannot work correctly.",
+	}
+	Eventually(func() bool {
+		_, err := r.resourceRetriever.GetKubegres()
+		if err != nil {
+			return false
+		}
+		return eventRecorderTest.CheckEventExist(expectedErrorEvent)
+
+	}, resourceConfigs.TestTimeout, resourceConfigs.TestRetryInterval).Should(BeTrue())
+}
+
+func (r *StandByTest) thenPodsStatesShouldBe(nbrePrimary, nbreReplicas int) bool {
+	return Eventually(func() bool {
+
+		pods, err := r.resourceRetriever.GetKubegresResources()
+		if err != nil && !apierrors.IsNotFound(err) {
+			log.Println("ERROR while retrieving Kubegres pods")
+			return false
+		}
+
+		if pods.AreAllReady &&
+			pods.NbreDeployedPrimary == nbrePrimary &&
+			pods.NbreDeployedReplicas == nbreReplicas {
+
+			time.Sleep(resourceConfigs.TestRetryInterval)
+			log.Println("Deployed and Ready StatefulSets check successful")
+			return true
+		}
+
+		return false
+
+	}, resourceConfigs.TestTimeout, resourceConfigs.TestRetryInterval).Should(BeTrue())
+}
+
+func (r *StandByTest) thenCronJobExistsWithSpec(expectedConfigMapName,
+	expectedBackupSchedule,
+	expectedBackupPvcName,
+	expectedBackupVolumeMount string) bool {
+
+	return Eventually(func() bool {
+
+		kubegresResources, err := r.resourceRetriever.GetKubegresResources()
+		if err != nil && !apierrors.IsNotFound(err) {
```

Comment:
I've seen that is repeated in multiple places, therefore correct.
But it looks "strange"/"incorrect" until you look at the implementation. 
The strange incorrect part is: what happens if `err!=nil` and it is a Not Found error? Well the implementation wipes the err and sets it to `nil`. Professional, XD, very professional XD.

Reply:
good one.